crates/utils/re_video/src/demux/mod.rs

```diff
@@ -509,6 +507,43 @@ impl VideoDataDescription {
         self.samples.num_elements()
     }
 
+    /// Duration of the video.
+    ///
+    /// For videos that may still grow this is an estimate of the video length so far.
+    /// (we don't know for sure how long the last sample is going to be)
+    pub fn duration(&self) -> Option<std::time::Duration> {
+        match &self.update_type {
+            VideoUpdateType::NoUpdates { duration } => Some(duration.duration(self.timescale?)),
+
+            VideoUpdateType::Stream { .. } => {
+                if self.samples.num_elements() < 2 {
+                    return None;
+                }
+
+                // TODO(#10090): This is only correct because there's no b-frames on streams right now.
+                // If there are b-frames determining the last timestamp is a bit more complicated.
+                let first_sample = self.samples.front()?;
+                let last_sample = self.samples.back()?;
+                let timescale = self.timescale?;
+
+                // Estimate length of the last sample:
+                let second_to_last_sample = self
+                    .samples
+                    .get(self.samples.next_index().saturating_sub(2))?;
+                let estimated_average_sample_duration = (last_sample.presentation_timestamp
+                    - second_to_last_sample.presentation_timestamp)
+                    .duration(timescale)
+                    / (self.num_samples() - 1) as u32;
```

Comment:
This looks VERY buggy

```suggestion
                let estimated_average_sample_duration =
                    (last_sample.presentation_timestamp - first.presentation_timestamp).duration(timescale)
                    / (last_sample.frame_nr - first.frame_nr);
```